boards: lilygo: ttgo_tbeam: remove not-needed kconfig opts

fmoessbauer opened the pull request:
The boards defconfig enabled CONFIG_(GPIO|MFD|REGULATOR), which are not stricly needed if the corresponding node is not enabled in the device tree. These options also enable many drivers, which are problematic on the mcuboot bootloader target. By dropping them, we restore the sysboot target (that uses mcuboot).

fmoessbauer committed:
boards: lilygo: ttgo_tbeam: remove not-needed kconfig opts

The boards defconfig enabled CONFIG_(MFD|REGULATOR), which are not
stricly needed if the corresponding node is not enabled in the device
tree. These options also enable many drivers, which are problematic on
the mcuboot bootloader target. By moving them to the Kconfig, we can
make them depend on !MCUBOOT which restores the sysboot target.

Signed-off-by: Felix Moessbauer <felix.moessbauer@gmail.com>

fmoessbauer commented:
cc @lfelten

lfelten commented:
Hi @fmoessbauer,
the TBeam uses the AXP2102 regulator in ttgo_tbeam_esp32_procpu.dts, it sets the voltages for DCDC1, ALDO2 and ALDO3. I have no information what the defaults for those voltages are.
You can disable the driver in the config if the default voltages are acceptable.

fmoessbauer commented:
Hi @lfelten , by having these options unconditionally enabled, the mcu bootloader cannot be built, because the regulator depends on `I2C` and that depends on `MULTITHREADING`, which is incompatible with the bootloader. For that, we need to remove these options at least for MCUBOOT. I also think that during stage 2 booting the regulator is not needed. After that, the application takes over and enables & configures it.

I'm wondering how this switch can be achieved. Probably we could move these options to the `Kconfig.sysbuild`, and make them `!MCUBOOT`. How about that? How it this handled on other boards?

fmoessbauer commented:
The CI error is unrelated and due to 92ebb2eb69f035605769da5b647edd2d0d83e19c.

fmoessbauer commented:
@lfelten I now changed the config options to depend on `!MCUBOOT` as during the first boot stage we simply cannot use multithreading (this flag was missing, fixed in #89392) and by that no ESP32 I2C driver can be used. I further checked that this works on the tbeam board (the second stage then activates the regulator and initializes the devices).
